include/glow/Importer/ProtobufLoader.h

```diff
@@ -181,8 +189,52 @@ class ProtobufLoader {
   /// \returns the Placeholder for the external output with \p name.
   /// \pre outputVarsByName_.find(name) != outputVarsByName_.end()
   llvm::Expected<Placeholder *> getOutputByName(llvm::StringRef name) const;
+
+  /// \returns True if the operator with name \p typeName having input node
+  /// list as \p inputs is constant foldable.
+  bool isConstantFoldable(llvm::SmallVector<NodeValue, 4> &inputs,
+                          std::string typeName) const;
 };
 
+/// \returns success if the folding of operator \p op in the loader
+/// \p pLoader is successful. The folding utility uses temporary
+/// loader \p loaderCF, and associated temporary function \p pFuncCF.
+template <class LoaderType, class OpType>
+llvm::Error constantFoldInLoader(Function *pFuncCF, LoaderType &loaderCF,
+                                 LoaderType *pLoader, const OpType &op) {
+  PlaceholderBindings bindingCF;
+  std::vector<Tensor *> pTensorVec;
+  Module *pModCF = pFuncCF->getParent();
```

Comment:
Please add some comments explaining what the following two loops are doing.

Reply:
done.